frontend/src/components/AccountDetailsEditContainer.vue

```diff
@@ -0,0 +1,225 @@
+            :loading="disableInputShowSpinner"
+            @click="deleteAccount()"
+          >
+            Löschen
+          </ButtonComponent>
+        </div>
+      </template>
+    </ContainerComponent>
+  </div>
+</template>
+<script setup lang="ts">
+import { ref } from 'vue';
+import TextInputComponent from './TextInputComponent.vue';
+import ButtonComponent from './ButtonComponent.vue';
+import ContainerComponent from './ContainerComponent.vue';
+import { IUser, UserClass } from '../../../model/user/IUser';
+import { AccountRestInterface } from '../restInterfaces/AccountRestInterface';
+
+const props = defineProps({
+  /**
+   * Whether the delete button is displayed
+   */
+  showadminview: {
+    type: Boolean,
+    required: false,
+    default: false,
+  },
+  /**
+   * The user to display
+   */
+  user: {
+    type: Object as () => Partial<IUser>,
+    required: true,
+  },
+});
+
+const nameField = ref(props.user.name);
+const mailField = ref(props.user.mail);
+const pwField = ref('');
+const pwFieldControl = ref('');
+const disableInputShowSpinner = ref(false);
+const deleted = ref(false);
+const displayError = ref(false);
+const displaySuccess = ref(false);
+let originalUser = props.user as IUser;
+const isAdmin = ref(originalUser.type === UserClass.ADMIN);
+/**
+ * toggles the checkbox
+ */
+function toggleCheckBox(): void {
+  if (!disableInputShowSpinner.value) {
+    isAdmin.value = !isAdmin.value;
+  }
+}
+
+/**
+ * Updates Information from the User with the entered information
+ */
+async function updateInformation(): Promise<void> {
+  if (deleted.value) {
+    return;
+  }
+  displaySuccess.value = false;
+  displayError.value = false;
+  disableInputShowSpinner.value = true;
+
+  const updateUser: Partial<IUser> = { id: originalUser.id };
+
+  if (nameField.value !== '') {
+    updateUser.name = nameField.value;
+  }
+  if (mailField.value !== '') {
+    updateUser.mail = mailField.value;
+  }
+  if (pwField.value !== '') {
+    if (pwField.value === pwFieldControl.value) {
+      updateUser.password = pwField.value;
+    } else {
+      displayError.value = true;
+      disableInputShowSpinner.value = false;
+      return;
+    }
+  }
+  if (props.showadminview) {
+    updateUser.type = isAdmin.value ? UserClass.ADMIN : UserClass.REGULAR;
+  }
+  await AccountRestInterface.updateAccount(updateUser);
+  disableInputShowSpinner.value = false;
+  displaySuccess.value = true;
+}
+/**
+ * deletes a account
+ */
+async function deleteAccount(): Promise<void> {
+  disableInputShowSpinner.value = true;
+  if (originalUser.id === null) {
+    console.log('user id is empty - should not happen');
+    return;
+  }
+  await AccountRestInterface.deleteAccount(originalUser.id as number);
+  disableInputShowSpinner.value = false;
+  deleted.value = true;
+}
+/**
+ * reseting this component
+ */
+async function reset(): Promise<void> {
+  if (deleted.value) {
+    return;
+  }
+  window.location.reload();
```

Comment:
This isnt a good solution. Reloading should be avoided in a single page application. Use the given user and reset all fields

Reply:
i dont see a better solution, since the last reset option i found is for vue2 (see: https://github.com/vuejs/vue/issues/702)

Reply:
I think the better option here is just to reset all the fields to the values of the originalUser and empty the password field, since we still got the reference to it in the user prop

Reply:
i still dont see how that would work since the input field is capsulated in a component. You may could push it or write how the code would look like?

Reply:
Shouldnt something like
`nameField.value = props.user.name` (similar for the others)
work?

Reply:
that would work in javascript (stackoverflow: `document.getElementById('MyId').value = newVal`) right? in vue according to my research i would need to have access to the v-model variable and set these to the given value, but i cant access these it component (i know that vue compiles to javascript, but i cant access the document variable in the `<script>` section)

Reply:
Ah, I see, because we can't override the value of TextInputComponents after we created them, at least right now - however, we could make our TextInputComponent compatible with v-model, but that'd change a fair bit of our existing work, since all TextInputComponents would need to be changed, although they'd fit in better with "regular vue" after I'd imagine - Opinions? (Also @Kr0nox)

Reply:
Status update: We decided to make the TextInputComponent v-model compatible. Updating this soon:tm:

Reply:
Status update: v-model is complete, this is now working without reloading the page